### ISSUE TITLE
App breaks upgrading from 0.4.54 to 0.4.55

### ISSUE THREAD

**Issue by arthurgailes** [brillout/vite-plugin-ssr#555]
### Description

My app starts fine in dev mode as of 0.4.54. In 0.4.55 and 0.4.59 (latest to-date), I get the following message and an exit code. There are a lot of changes between the two version, and I'm having a touch time seeing where this is coming from.

### Error Message + Error Stack

```shell
Error: [vite-plugin-ssr@0.4.55][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Reach out at https://github.com/brillout/vite-plugin-ssr/issues/new or https://discord.com/invite/qTq92FQzKb and include this error stack (the error stack is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as 
it makes vite-plugin-ssr more robust.
     at determineFsAllowList ([...]\Documents\sites\aei-js-monorepo\node_modules\.pnpm\vite-plugin-ssr@0.4.55_33oe7thcih3clfsuwaxtnji33q\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\devConfig.js:87:24)
     at configResolved ([...]\Documents\sites\aei-js-monorepo\node_modules\.pnpm\vite-plugin-ssr@0.4.55_33oe7thcih3clfsuwaxtnji33q\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\devConfig.js:41:23)
     at async Promise.all (index 6)
     at async resolveConfig (file:///[...]/Documents/sites/aei-js-monorepo/node_modules/.pnpm/vite@3.2.4_@types+node@18.0.0/node_modules/vite/dist/node/chunks/dep-67e7f8ab.js:62575:5)
     at async createServer (file:///[...]/Documents/sites/aei-js-monorepo/node_modules/.pnpm/vite@3.2.4_@types+node@18.0.0/node_modules/vite/dist/node/chunks/dep-67e7f8ab.js:61641:20)
     at async CAC.<anonymous> (file:///[...]/Documents/sites/aei-js-monorepo/node_modules/.pnpm/vite@3.2.4_@types+node@18.0.0/node_modules/vite/dist/node/cli.js:707:24)  
 pnpm vite exited with code 1
```

**Comment by brillout** [comment]
Insert

```js
    console.log('fsAllow:', fsAllow);
    console.log('searchForWorkspaceRoot:', (0, utils_1.toPosixPath)((0, vite_1.searchForWorkspaceRoot)(process.cwd())));
```
before the line `\Documents\sites\aei-js-monorepo\node_modules\.pnpm\vite-plugin-ssr@0.4.55_33oe7thcih3clfsuwaxtnji33q\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\devConfig.js:87:24` and report back the output.

**Comment by arthurgailes** [comment]
FWIW fsAllow is my workspace root and searchForWorkspaceRoot is the project root. 

```
fsAllow: [ 'C:/[...]/Documents/sites/aei-js-monorepo' ]
searchForWorkspaceRoot: C:/[...]/Documents/sites/aei-js-monorepo/apps/hilton/vite
error when starting dev server:
Error: [vite-plugin-ssr@0.4.55][Bug] You stumbled upon a bug in vite-plugin-ssr's source code. Reach out at https://github.com/brillout/vite-plugin-ssr/issues/new or https://discord.com/invite/qTq92FQzKb and include this error stack (the error stack is usually enough to fix the problem). A maintainer will fix the bug (usually under 24 hours). Don't hesitate to reach out as it makes vite-plugin-ssr more robust.
    at determineFsAllowList (C:\[...]\Documents\sites\aei-js-monorepo\node_modules\.pnpm\vite-plugin-ssr@0.4.55_33oe7thcih3clfsuwaxtnji33q\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\devConfig.js:89:24)
    at configResolved (C:\[...]\Documents\sites\aei-js-monorepo\node_modules\.pnpm\vite-plugin-ssr@0.4.55_33oe7thcih3clfsuwaxtnji33q\node_modules\vite-plugin-ssr\dist\cjs\node\plugin\plugins\devConfig.js:41:23)
    at async Promise.all (index 6)
    at async resolveConfig (file:///C:/[...]/Documents/sites/aei-js-monorepo/node_modules/.pnpm/vite@3.2.4_@types+node@18.0.0/node_modules/vite/dist/node/chunks/dep-67e7f8ab.js:62575:5)
    at async createServer (file:///C:/[...]/Documents/sites/aei-js-monorepo/node_modules/.pnpm/vite@3.2.4_@types+node@18.0.0/node_modules/vite/dist/node/chunks/dep-67e7f8ab.js:61641:20)
    at async CAC.<anonymous> (file:///C:/[...]/Documents/sites/aei-js-monorepo/node_modules/.pnpm/vite@3.2.4_@types+node@18.0.0/node_modules/vite/dist/node/cli.js:707:24)pnpm vite exited with code 1
```
When I repeated this step in 0.4.54, it broke and printed the following: 
```
fsAllow: [ 'C:/Users/arthur.gailes/Documents/sites/aei-js-monorepo' ]
error when starting dev server:
ReferenceError: vite_1 is not defined
```

**Comment by brillout** [comment]
Ok so `fsAllow` is correct while `searchForWorkspaceRoot` isn't (it should be equal to the `fsAllow` entry).

What's the output of following?

```
console.log('cwd', process.cwd());
console.log('searchForWorkspaceRoot 1', vite_1.searchForWorkspaceRoot(process.cwd());
console.log('searchForWorkspaceRoot 2', vite_1.searchForWorkspaceRoot(utils_1.toPosixPath(process.cwd()));
```

**Comment by arthurgailes** [comment]
cwd [...]\Documents\sites\aei-js-monorepo\apps\hilton\vite
searchForWorkspaceRoot 1 [...]\Documents\sites\aei-js-monorepo\apps\hilton\vite
searchForWorkspaceRoot 2 [...]/Documents/sites/aei-js-monorepo/apps/hilton/vite

**Comment by brillout** [comment]
Thanks for sponsoring btw.! And if you manage (I know it's hard) to convince your company to join the sponsorship program, that'd be rad. Thanks!

**Comment by arthurgailes** [comment]
This works well, thank you!

**Comment by arthurgailes** [comment]
I spoke too soon, this breaks the process entirely on linux (AWS Fedora) on line 87 :

```
    (0, utils_1.assert)(process.platform === 'win32' || // searchForWorkspaceRoot() seems buggy on Windows https://github.com/brillout/vite-plugin-ssr/issues/555
        fsAllow.map(utils_1.toPosixPath).includes((0, utils_1.toPosixPath)((0, vite_1.searchForWorkspaceRoot)(process.cwd()))));
```

Reverting back to 0.4.54 fixes it again

**Comment by brillout** [comment]
@arthurgailes What's the output of the following?

```
console.log('cwd', process.cwd());
console.log('searchForWorkspaceRoot', vite_1.searchForWorkspaceRoot(process.cwd());
```

**Comment by arthurgailes** [comment]
The exact same string.

cwd [workspaceRoot]/apps/hilton/vite
searchForWorkspaceRoot [workspaceRoot]/apps/hilton/vite

**Comment by brillout** [comment]
Fix released in `0.4.64`. Should be good now 😁.

**Comment by arthurgailes** [comment]
Works for me, thank you.
